topo/node/drivenets/drivenets.go

```diff
@@ -0,0 +1,307 @@
+	}
+}
+
+func (n *Node) Delete(ctx context.Context) error {
+	if n.Impl.Proto.Model != modelCdnos {
+		return fmt.Errorf("unknown model")
+	}
+	return n.cdnosDelete(ctx)
+}
+
+func (n *Node) cdnosDelete(ctx context.Context) error {
+	cs, err := clientFn(n.RestConfig)
+	if err != nil {
+		return err
+	}
+	return cs.CdnosV1alpha1().Cdnoss(n.Namespace).Delete(ctx, n.Name(), metav1.DeleteOptions{})
+}
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	log.Info("ResetCfg is a noop.")
+	return nil
+}
+
+func (n *Node) ConfigPush(context.Context, io.Reader) error {
+	return status.Errorf(codes.Unimplemented, "config push is not implemented using gNMI to configure device")
+}
+
+func (n *Node) GenerateSelfSigned(context.Context) error {
+	return status.Errorf(codes.Unimplemented, "certificate generation is not supported")
+}
+
+func cdnosDefaults(pb *tpb.Node) *tpb.Node {
+	if pb.Config == nil {
+		pb.Config = &tpb.Config{}
+	}
+	if pb.Config.Image == "" {
+		pb.Config.Image = "registry.dev.drivenets.net/devops/cdnos:latest"
+	}
+	if pb.Config.InitImage == "" {
+		pb.Config.InitImage = node.DefaultInitContainerImage
+	}
+	if len(pb.GetConfig().GetCommand()) == 0 {
+		pb.Config.Command = []string{"/define_notif_net.sh"}
+	}
+	if pb.Config.EntryCommand == "" {
+		pb.Config.EntryCommand = fmt.Sprintf("kubectl exec -it %s -- /bin/bash", pb.Name)
+	}
+	if pb.Config.Cert == nil {
+		pb.Config.Cert = &tpb.CertificateCfg{
+			Config: &tpb.CertificateCfg_SelfSigned{
+				SelfSigned: &tpb.SelfSignedCertCfg{
+					CommonName: pb.Name,
+					KeySize:    2048,
+				},
+			},
+		}
+	}
+	if pb.Constraints == nil {
+		pb.Constraints = map[string]string{}
+	}
+	if pb.Constraints["cpu"] == "" {
+		pb.Constraints["cpu"] = "0.5"
+	}
+	if pb.Constraints["memory"] == "" {
+		pb.Constraints["memory"] = "1Gi"
+	}
+	if pb.Labels == nil {
+		pb.Labels = map[string]string{}
+	}
+	if pb.Labels["vendor"] == "" {
+		pb.Labels["vendor"] = tpb.Vendor_DRIVENETS.String()
+	}
+
+	// Always explicitly specify that cdnos is a DUT, this cannot be overridden by the user.
+	pb.Labels[node.OndatraRoleLabel] = node.OndatraRoleDUT
+
+	if pb.Services == nil {
+		pb.Services = map[uint32]*tpb.Service{
+			// https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=gnmi
+			9339: {
+				Names:  []string{"gnmi", "gnoi", "gnsi"},
+				Inside: 9339,
+			},
+			9340: {
+				Names:  []string{"gribi"},
+				Inside: 9340,
+			},
+			830: {
+				Names:  []string{"netconf"},
+				Inside: 830,
+			},
+		}
+	}
+	return pb
+}
+
+func init() {
+	node.Vendor(tpb.Vendor_DRIVENETS, New)
+}
+
+func (n *Node) CreateConfig(ctx context.Context) (*corev1.Volume, error) {
+	pb := n.Proto
+	var data []byte
+	switch v := pb.Config.GetConfigData().(type) {
+	case *tpb.Config_File:
+		var err error
+		data, err = os.ReadFile(filepath.Join(n.BasePath, v.File))
+		if err != nil {
+			return nil, err
+		}
+	case *tpb.Config_Data:
+		data = v.Data
+	}
+	if data != nil {
```

Comment:
can flip this conditional to reduce indentation:

```
if data == nil {
    return nil, nil
}
...
```

Reply:
ok, i modifed the code accordingly